src/bitcoind.cpp

```diff
@@ -184,9 +184,13 @@ static bool AppInit(NodeContext& node)
             return false;
         }
 
-        node.kernel = std::make_unique<kernel::Context>();
-        if (!AppInitSanityChecks(*node.kernel))
-        {
+        auto result{kernel::Context::MakeContext()};
+        if (!result) {
+            InitError(util::ErrorString(result));
+            return InitError(strprintf(_("Initialization kernel sanity check failed. %s is shutting down."), PACKAGE_NAME));
```

Comment:
This will trigger two error dialogs in the GUI, one after the other.
Maybe better to concatenate the errors and call `InitError()` only once?